lib/galaxy/datatypes/text.py

```diff
@@ -573,3 +573,37 @@ def sniff_prefix(self, file_prefix):
         False
         """
         return file_prefix.startswith("IQ-TREE")
+
+class Top(Text):
+    """Class describing an gromacs topology file"""
+    file_ext = "top"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Topology file"
+            dataset.blurb = nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'text/top'
+
+    def sniff_prefix(self, file_prefix):
+        """
```

Comment:
This one as well. The sniffer seems to be not strong enough. Everything with `atoms` will be recognized as Top file.
Just subclass from text in the datatypes_conf.xml file instead.
The downside is that users during upload need to specify the file type, which I think is ok in this case.